src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,527 @@
+            # X is horizontal axis, Y is vertical axis
+            pxs = milling_settings.pixelSize.value
+            probe_size = max(pxs)
+            overlap = [1 - (pxs[0] / probe_size), 1 - (pxs[1] / probe_size)]  # Always >= 0
+
+            if self._future._task_state == CANCELLED:
+                raise CancelledError()
+
+            hori_dim = (milling_settings.roi.value[2] - milling_settings.roi.value[0]) * \
+                       milling_settings.pixelSize.value[0]
+            vert_dim = (milling_settings.roi.value[3] - milling_settings.roi.value[1]) * \
+                       milling_settings.pixelSize.value[1]
+            dims = [hori_dim, vert_dim]
+
+            hori_t = ((milling_settings.roi.value[2] - milling_settings.roi.value[0]) / 2 - 0.5) * \
+                     milling_settings.pixelSize.value[0]
+            vert_t = ((milling_settings.roi.value[3] - milling_settings.roi.value[1]) / 2 - 0.5) * \
+                     milling_settings.pixelSize.value[1]
+            trans = [hori_t, vert_t]
+
+            waiting_log_time = milling_settings.duration.value + self.estimate_drift_time(milling_settings)
+
+            logging.debug(f"Milling setting: {milling_settings.name.value} for feature: {site.name.value}")
+            logging.info(
+                f"Milling a rectangle of dimensions: {dims} m which is {trans} m away from the center will "
+                f"take approximately {waiting_log_time} s")
+
+            if milling_settings.dcRoi.value == UNDEFINED_ROI:
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+
+                self._future.running_subf = mill_rectangle(milling_settings.roi.value, self._scanner,
+                                                           iteration=self._iterations[setting_nb],
+                                                           duration=self._pass_duration[setting_nb],
+                                                           probe_size=probe_size, overlap=overlap)
+                try:
+                    self._future.running_subf.result()
+                    logging.debug(f"Milling finished for feature: {site.name.value}")
+                except Exception as exp:
+                    logging.error(
+                        f"Milling setting: {milling_settings.name.value} for feature: {site.name.value} failed: {exp}")
+                    raise
+
+            else:
+                for itr in range(self._iterations[setting_nb]):
+                    if self._future._task_state == CANCELLED:
+                        raise CancelledError()
+
+                    # restore the milling current
+                    self._restore_beam_shift(milling_settings.current.value)
+
+                    self._future.running_subf = mill_rectangle(milling_settings.roi.value, self._scanner,
+                                                               iteration=1,
```

Comment:
Please just write `self._iterations[setting_nb]` , to make it clear that it does nothing really special, and especially make it easy to change to a different number of iterations if we want to.